src/gui/KM_InterfaceMapEditor.pas

```diff
@@ -841,7 +841,22 @@ procedure TKMapEdInterface.MouseUp(Button: TMouseButton; Shift: TShiftState; X,Y
                   TKMUnit(gMySpectator.Selected).SetPosition(gGameCursor.Cell);
 
                 if gMySpectator.Selected is TKMUnitGroup then
-                  TKMUnitGroup(gMySpectator.Selected).Position := gGameCursor.Cell;
+                begin
+                  if ssShift in gGameCursor.SState then
+                  begin
+                    if (gTerrain.UnitsHitTest(gGameCursor.Cell.X, gGameCursor.Cell.Y) <> nil)
+                    or (gHands.HousesHitTest(gGameCursor.Cell.X, gGameCursor.Cell.Y) <> nil) then
+                      TKMUnitGroup(gMySpectator.Selected).MapEdOrder.Order := ioAttackPosition
+                    else
+                      TKMUnitGroup(gMySpectator.Selected).MapEdOrder.Order := ioSendGroup;
+                    TKMUnitGroup(gMySpectator.Selected).MapEdOrder.Pos.Loc.X := gGameCursor.Cell.X;
+                    TKMUnitGroup(gMySpectator.Selected).MapEdOrder.Pos.Loc.Y := gGameCursor.Cell.Y;
+                    TKMUnitGroup(gMySpectator.Selected).MapEdOrder.Pos.Dir := TKMUnitGroup(gMySpectator.Selected).Direction;

```

Comment:
Too many repetitions of `TKMUnitGroup(gMySpectator.Selected)`. Can we replace it with a temp variable?

Also please add comments about the code (what it does)